tests/integration/test_api.py

```diff
@@ -383,6 +389,39 @@ def test_annotation_event(client, monkeypatch, httpserver):
     assert result.status_code == 200
 
 
-def test_display_insight_prediction_for_products(client, mocker):
+def test_insight_prediction(client, mocker):
```

Comment:
the name is not really descriptive.
More over you don't need the mocker parameter if you do not use it in your test.

```suggestion
def test_prediction_collection_no_filter(client):
```